src/utils/fileValidation.js

```
@@ -0,0 +1,353 @@
+export const getFileExtension = (filename) => {
+  return filename.slice((filename.lastIndexOf('.') - 1 >>> 0) + 2)
+}
+
+export const isValidExtension = (filename) => {
+  const extension = getFileExtension(filename).toLowerCase()
+  return ALLOWED_EXTENSIONS.some(allowed => allowed.toLowerCase() === `.${extension}`)
+}
+
+export const containsSuspiciousPatterns = (filename) => {
+  const suspiciousPatterns = [
+    /[<>:"|?*\x00-\x1f]/,  // Invalid filename characters
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)